### DISCUSSION TITLE
Setup does not work with two RWB inverters

### DISCUSSION THREAD

**Discussion by suelm** [WillCodeForCats/solaredge-modbus-multi#586] [Q&A] [answered]
One Master RWB and one Follower RWB. These use RS485-2 for the follower and not RS485-1. If I specify in the configuration that I have two inverters, the setup fails. If I specify only one inverter, it is found, but with the wrong values.

![Image (1)](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/33831840/358d6498-0b76-4a72-8209-9628470ad1c8)

**Comment by suelm** [comment]
Unfortunately, I cannot find a configuration error.
The first inverter has ID 1 and the second inverter has ID 2. The inverters communicate via RS485-2.
The inline meter is connected to RS485-1.
There is also the backup interface and battery.

If I specify only one inverter in the configuration, it finds the battery and inline meter.
If I enter two inverters, the setup can no longer find anything. 

What can I do?

**Comment by WillCodeForCats** [comment]
Turn on debug to log why it doesn't see ID 2.

**Reply by suelm** [reply]
[home-assistant_solaredge_modbus_multi_2024-04-23T06-21-13.032Z.log](https://github.com/WillCodeForCats/solaredge-modbus-multi/files/15072286/home-assistant_solaredge_modbus_multi_2024-04-23T06-21-13.032Z.log)

**Reply by WillCodeForCats** [reply]
There is no response from the next inverter.

After this line it will go to the next inverter:
```
2024-04-23 08:20:58.909 DEBUG (MainThread) [custom_components.solaredge_modbus_multi.hub] I1B3: Battery 3 not usable (rating <=0)
```
But there is never any response and eventually it times out:
```
2024-04-23 08:21:10.915 DEBUG (MainThread) [custom_components.solaredge_modbus_multi] Finished fetching SolarEdge Coordinator data in 12.780 seconds (success: False)
```

**Reply by suelm** [reply]
Yes, that's how I see it too. But the inverter is there and communicates correctly with the first inverter. Do you have any idea what I could do? I would love to use the integration.

**Reply by WillCodeForCats** [reply]
I added some more log messages in [Release v2.4.14-pre.3](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v2.4.14-pre.3) for the setup process.

**Reply by suelm** [reply]
Here is the log from the new version
[home-assistant_solaredge_modbus_multi_2024-04-27T08-06-14.125Z.log](https://github.com/WillCodeForCats/solaredge-modbus-multi/files/15137595/home-assistant_solaredge_modbus_multi_2024-04-27T08-06-14.125Z.log)

**Reply by WillCodeForCats** [reply, marked as answer]
The last thing it logs is `Looking for inverter at 192.168.1.252 ID 2` which means that ID 2 never responded.

I can't fix that in the integration, you would need to verify the inverters have their ID set correctly.

Screenshots for the ID settings are here: https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration

**Reply by suelm** [reply]
I have checked it again. It was indeed my mistake, a different ID was displayed via SetApp than via the monitoring. 
Thank you very much for your help.
